Tests/Integration/Components/CategoryResolverTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * (c) shopware AG <info@shopware.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ShopwarePlugins\Connect\Tests\Integration\Components;
+
+use Shopware\CustomModels\Connect\ProductToRemoteCategory;
+use ShopwarePlugins\Connect\Components\CategoryResolver\AutoCategoryResolver;
+use ShopwarePlugins\Connect\Components\ConfigFactory;
+use ShopwarePlugins\Connect\Tests\DatabaseTestCaseTrait;
+
+class CategoryResolverTest extends \PHPUnit_Framework_TestCase
+{
+    use DatabaseTestCaseTrait;
+
+    private $manager;
+    private $config;
+    private $categoryRepo;
+    private $categoryResolver;
+
+    public function setUp()
+    {
+        $this->manager = Shopware()->Models();
+        $this->config = ConfigFactory::getConfigInstance();
+        $this->categoryRepo = $this->manager->getRepository('Shopware\Models\Category\Category');
+
+        $this->categoryResolver = new AutoCategoryResolver(
+            $this->manager,
+            $this->categoryRepo,
+            $this->manager->getRepository('Shopware\CustomModels\Connect\RemoteCategory'),
+            $this->config,
+            $this->manager->getRepository(ProductToRemoteCategory::class)
+        );
+    }
+
+    public function testStoreRemoteCategories()
+    {
+        $this->manager->getConnection()->executeQuery('DELETE FROM s_plugin_connect_categories');
+        $this->manager->getConnection()->executeQuery('DELETE FROM s_plugin_connect_product_to_categories');
```

Comment:
why you need both delete lines ? Integration tests already work with transactions, so it should be a problem only to insert records.

Reply:
yes but the legacy tests don't always use transaction they run before